content-elements/form/form-pin/form-pin.js

```diff
@@ -0,0 +1,179 @@
+  },
+
+  validateInput() {
+    if (this.bsiInputElement.value.length != this.maxlength) {
+      this.requiredErrorElement.style.display = "block";
+    } else {
+      this.requiredErrorElement.style.display = "none";
+    }
+  },
+
+  initPinNumberFields() {
+    this.bsiInputElement = this.$root.querySelector('.bsi-form-field-input-original');
+    let containerDiv = this.$root.querySelector('.bsi-form-pin-element');
+
+    const maxLength = this.bsiInputElement.getAttribute('maxlength');
+    if (maxLength == null) {
+      this.maxlength = 6;
+    } else {
+      this.maxlength = this.bsiInputElement.getAttribute('maxlength');
+    }
+
+    for (let i = 0; i < this.maxlength; i++) {
+      this._initPinNumberField(containerDiv, i);
+    }
+  },
+
+  _initPinNumberField(containerDiv, i) {
+    let div = document.createElement('div');
+    let label = document.createElement('label');
+    let pinInput = document.createElement('input');
+    
+    div.classList = 'input-wrapper';
+    label.classList = 'form-label';
+    label.innerHTML = (i + 1) + ".";
+    pinInput.classList = 'bsi-form-field-input form-control bsi-form-field-input pin';
+    
+    pinInput.setAttribute('x-init', 'initFormPinInput');
+    pinInput.setAttribute('required', 'true');
+
+    div.appendChild(label);
+    div.appendChild(pinInput);
+    containerDiv.appendChild(div);
+  },
+  
+  initFormPinInput() {
+    let inputPin = this.$el;
+    inputPin.setAttribute('inputmode', 'numeric');
+    inputPin.setAttribute('pattern', "[0-9]");
+
+    inputPin.addEventListener('focusin', (e) => {
+      inputPin.setAttribute('old', inputPin.value);
+    });
+
+    inputPin.addEventListener('keydown', (e) => {
+      if (e.key ==='Backspace' && !inputPin.value) {
+        this._autoFocusPreviousPinInput();
+      }
+    });
+
+    inputPin.addEventListener('input', (e) => {
+      const inputPinList = this.$root.querySelectorAll('input.pin');
+      this._cleanUp();
+      this.bsiInputElement.value = Array.from(inputPinList).reduce((result, input) => {
+        return result + input.value;
+      }, "");
+      if (this.$root.classList.contains('auto-submit')) {
+        this._autoSubmitIfFilledIn();
+      }
+      this._autoFocusNextPinInput();
+    });
+  },
+
+  _autoFocusNextPinInput() {
+    let inputPin = this.$el;
+    let nextWrapper = inputPin.parentNode.nextElementSibling;
+    if (inputPin.value && !this._isLastPinElement()) {
+      this._autoFocus(nextWrapper);
+    }
+  },
+
+  _autoFocusPreviousPinInput() {
+    let inputPin = this.$el;
+    let previousWrapper = inputPin.parentNode.previousSibling;
+    if (previousWrapper != null) {
+      this._autoFocus(previousWrapper);
+    }
+  },
+
+  _autoFocus(wrapper){
+    var nextPinInput = wrapper.querySelector('input.pin');
+      if (nextPinInput) {
+        nextPinInput.focus();
+      }
+  },
+
+  _autoSubmitIfFilledIn() {
+    let form = this.$root.querySelector('.formular-pin');;
```

Comment:
Außerdem gibt es .formular-pin nicht mehr, oder?
Ich wäre für `this.$root.closest('form')`

Reply:
Ich habe im letzten Commit das Formular mit dem Pin Element zu einem kombined. Deshalb gibt es das schon irgendwie noch, nur, dass es jetzt ein Teil vom Pin Element ist.